src/FakeItEasy/Configuration/ArgumentCollection.cs

```diff
@@ -25,6 +25,17 @@ public class ArgumentCollection
         ///   Initializes a new instance of the <see cref = "ArgumentCollection" /> class.
         /// </summary>
         /// <param name = "arguments">The arguments.</param>
+        /// <param name = "method">The method.</param>
+        [DebuggerStepThrough]
+        public ArgumentCollection(object[] arguments, MethodInfo method)
+            : this(arguments, GetArgumentNames(method))
+        {
+        }

```

Comment:
Probably to create a custom implementation of `IInterceptedFakeObjectCall`. Essentially an external `CastleInvocationCallAdapter`.